File: src/test/java/ch/softappeal/yass/serialize/contract/Node.java
package ch.softappeal.yass.serialize.contract;

import ch.softappeal.yass.util.Tag;

import java.io.Serializable;

@Tag(30) public final class Node implements Serializable {

  private static final long serialVersionUID = 1L;

  // needed for coverage tests; these fields won't be serialized
  public transient int transientInt;
  public static int staticInt;

  @Tag(0) public int id;
  @Tag(1) public Node link;

  public Node(final int id) {
    this.id = id;
  }

  public Node() {
    // empty
  }

}

File: kotlin/yass/test/ch/softappeal/yass/kotlindir/serialize/contract/Node.kt
package ch.softappeal.yass.kotlindir.serialize.contract

import ch.softappeal.yass.Tag

import java.io.Serializable

@Tag(50)
class Node : Serializable {
    @Tag(1)
    var id: Int = 0
    @Tag(2)
    var link: Node? = null

    constructor(id: Int) {
        this.id = id
    }

    constructor()

    companion object {
        private const val serialVersionUID = 1L
        var staticInt: Int = 0
    }
}
